fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+# (c) Copyright IBM Corp. 2010, 2019. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+"""Function implementation"""
+
+import logging
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+import sys
+import os
+import json
+import grpc
+import re
+import inspect
+from google.protobuf.json_format import MessageToDict
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements generic wrapper for the gRPC client on the Resilient platform"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_grpc_interface", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_grpc_interface", {})
+
+    def _get_interface_file_names(self, base_dir=None, files_dir=None):
```

Comment:
* We need to avoid defining functions here and accessing them using `self.` (unless they are `static` functions)
* For functions used within the `_function_grpc_function` we should define them within the function
* This avoids conflicts when this Function is called multiple times from the Resilient Appliance

Reply:
ok